### ISSUE TITLE
[Container] Author can choose whichever color even if swatches only option is enabled

### ISSUE THREAD

**Issue by jakublamprecht** [adobe/aem-core-wcm-components#740]
## Bug Report

**Current Behavior**
Currently in the Container component we can choose to enable authors to chose background colors only from defined swatches (which is controlled by a checbox on container component policy).
The problem is that if we do that, in configuration dialog of a component we can only choose the swatches, but if we choose one, then we can freely change the color values in selected swatch, hence select any color we want.

**Expected behavior/code**
Author can choose the color only from predefined swatches

**Environment**
- Core Components 2.6.0

**Possible Solution**
To prevent this from happening it would be best to disable the text input which shows the current value of selected swatch.

**Additional context / Screenshots**
![swatches](https://user-images.githubusercontent.com/18114264/65036448-c0af4f80-d94b-11e9-8c9b-786984178144.gif)

**Comment by kotisyamala** [comment]
Hi @jakublamprecht 
If we disable the text field of the background color what should happen if the author selects swatches only in design, but didn't add any of the swatch colors?
cc: @richardhand

**Comment by richardhand** [comment]
Disabling the text field is a no go, as it would prevent the color value with submitting with the dialog form and requires hacking the internals of the color field.

The `variant={Boolean}swatch` property of the colorfield seems to help us out here, although I didn't try it.
```
/**
       * Enumeration for colorfield variant:
       *
       * default
       *    Use ColorField as a formfield
       * swatch
       *    Use a simple swatch as ColorField
       */
      - variant (String) = 'default'
```

**Comment by jakublamprecht** [comment]
Oh yeah, that seems right, forgot about that. I agree that another solution has to be applied here. The one that @richardhand suggests would probably be the cleanest.

**Comment by richardhand** [comment]
It seems to work well, when in fullscreen mode:
<img width="749" alt="Screenshot 2019-10-08 at 11 02 47" src="https://user-images.githubusercontent.com/25616660/66382436-7a359980-e9bb-11e9-87d8-25adb49b7ef5.png">

Unfortunately there seems to be a bug in the collision detection of the overlay in the default mode:
<img width="304" alt="Screenshot 2019-10-08 at 11 02 56" src="https://user-images.githubusercontent.com/25616660/66382446-7d308a00-e9bb-11e9-8519-917d311e7b11.png">

**Comment by richardhand** [comment]
The code for this issue is merged to master and will be available in the upcoming [2.8.0 release](https://github.com/adobe/aem-core-wcm-components/milestone/18).
